pkg/terra/client.go

```diff
@@ -56,66 +54,29 @@ func NewClient(spec OCR2Spec, lggr Logger) (*Client, error) {
 		WithTxConfig(app.MakeEncodingConfig().TxConfig)
 
 	return &Client{
-		close:              make(chan struct{}),
 		codec:              codec.NewLegacyAmino(),
 		chainID:            spec.ChainID,
 		clientCtx:          clientCtx,
 		cosmosRPC:          spec.CosmosURL,
-		httpClient:         &http.Client{Timeout: spec.HTTPTimeout},
```

Comment:
nit:
why get rid of the client? it seems cleaner than recreating the contexts each time?

Reply:
I'm thinking there might be other non-http.Client places we want to use that timeout and pulling it from an already constructed http.Client feels weird